### PR TITLE
Testing of the threaded_message_list_display_error change

### PR DESCRIPTION
**Pull request by philipwhiuk**
Had to fix a minor CCME. Might not be reproducible in main code. Makes the testing possible though.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/service/MailService.java [hunk 1]
```diff
@@ -342,7 +344,10 @@ private void reschedulePushers(boolean hasConnectivity, boolean doBackground) {
 
     private void setupPushers() {
         boolean pushing = false;
-        for (Account account : Preferences.getPreferences(MailService.this).getAccounts()) {
+        List<Account> accounts =
+                new ArrayList<Account>(Preferences.getPreferences(MailService.this)
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/service/MailService.java]: This is fixing the problem at the wrong end. We need to make a defensive copy before returning the list in `Preferences.getAccounts()`

Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/service/MailService.java]: And we are. Which makes me think this didn't fix it anyway and it's more subtle. I'll revert this for now and try and reproduce it later.

Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/service/MailService.java]: `Collections.unmodifiableList()` only wraps the underlying list and prevents modifying the list when using the wrapped object. Changes to the underlying list are still reflected in the unmodifiable list.

Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/service/MailService.java]: D'oh. Okay. I'll rework it later...

**Pull request merged**